### PR TITLE
Use aws_signature as dependency

### PR DESCRIPTION
**Pull request by philss**
This replaces the need for implementing the AWS Signature v4 in this
project.
Also we don't need to keep the algo for querystrings because they aren't
in use.

### CODE REVIEW

File: lib/aws/signature.ex [hunk 1]
```diff
@@ -14,229 +10,19 @@ defmodule AWS.Signature do
...
-  # in requests for all services does not cause any issues.
-  defp add_content_hash(headers, body) do
-    [{"X-Amz-Content-SHA256", AWS.Util.sha256_hexdigest(body)} | headers]
-  end
-
-  # Add an `X-Amz-Security-Token` if credentials configurations are configured for it
-  defp add_security_token(headers, %AWS.Client{session_token: nil}), do: headers
-
-  defp add_security_token(headers, %AWS.Client{session_token: session_token}),
-    do: [{"X-Amz-Security-Token", session_token} | headers]
```
Comment (jonatanklosko) [inline lib/aws/signature.ex:101]: `:aws_signature` doesn't add this header, so we need to add it conditionally before the signature.

Reasoning: this is pretty much an arbitrary, optional header and it's not relevant to the signing process, so it should be added manually beforehand as any other header :)

Comment (philss) [reply lib/aws/signature.ex:101]: @jonatanklosko good catch! I didn't notice it.

File: lib/aws/signature.ex [hunk 2]
```diff
@@ -14,229 +10,19 @@ defmodule AWS.Signature do
...
+      client.access_key_id,
+      client.secret_access_key,
+      region,
+      client.service,
+      {{now.year, now.month, now.day}, {now.hour, now.minute, now.second}},
+      to_string(method),
+      url,
+      headers,
+      body
     )
```
Comment (jonatanklosko) [inline lib/aws/signature.ex:26]: We should pass `uri_encode_path: false` if the service is S3. But this should go together with the path escaping fix I mentioned offline.

Example: given S3 key `"my$key"`, it should be encode into URL like `"/my%24key"` before passing it to `sign_v4`. Then for S3 we disable the additional encoding and it stays as `"/my%24key"` when normalizing, but for any other services it's further encoded as `"/my%2524key"`.

Comment (philss) [reply lib/aws/signature.ex:29]: I think I got it right. Please check again.

Comment (jonatanklosko) [reply lib/aws/signature.ex:29]: Looks great! I think the change should also be reflected in the codegen package.

Comment (philss) [reply lib/aws/signature.ex:29]: https://github.com/aws-beam/aws-codegen/commit/31b38916de305fa0c6257ba123f21f1894793b2c

Comment (jonatanklosko) [reply lib/aws/signature.ex:29]: Awesome!

**Pull request merged**